### PR TITLE
More librgw zipper9

### PR DESCRIPTION
**Pull request by mattbenjamin**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by mattbenjamin** [comment]
@dang these are critical to fix master, have decoupled from qa task to get them merged asap

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -5110,6 +5110,11 @@ void RGWCopyObj::execute(optional_yield y)
   if (init_common() < 0)
     return;
 
+  /* XXX fixup for rgw_file call path */
+  if (! s->object->get_bucket()) {
+    s->bucket = src_object->get_bucket()->clone();
```
Comment (dang) [inline src/rgw/rgw_op.cc:5115]: Maybe also s->object->set_bucket(s->bucket->get()); ?

**Review by dang — COMMENTED** [review]
One question, otherwise looks good.

File: src/rgw/rgw_op.cc (hunk 1 above)
Comment (mattbenjamin) [reply src/rgw/rgw_op.cc:5115]: is that more correct?  will change

Comment (mattbenjamin) [reply src/rgw/rgw_op.cc:5115]: (i mean, it seems obviously better, didn't realize it would work)

Comment (mattbenjamin) [reply src/rgw/rgw_op.cc:5115]: ack, will add

**Review by dang — APPROVED** [review]

**Pull request merged**